js/private/npm_import.bzl

```diff
@@ -75,31 +75,44 @@ def is_windows_os(rctx):
     os_name = rctx.os.name.lower()
     return os_name.find("windows") != -1
 
-# TODO: handle npm packages with lifecycle hooks; src will be the output of the postinstall rule instead
 _NODEJS_PACKAGE_TMPL = \
-"""def nodejs_package():
+    """{maybe_load_run_binary}
+
+def nodejs_package():
     if "{link_package_guard}" != "." and native.package_name() != "{link_package_guard}":
         fail("The nodejs_package() macro loaded from {nodejs_package_bzl} may only be called in the '{link_package_guard}' package. Move the call to the '{link_package_guard}' package BUILD file.")
 
+    {maybe_lifecycle_hooks}
+
     _nodejs_package(
         name = "{namespace}__{bazel_name}",
-        src = "@{rctx_name}//:{dir}",
+        src = "{nodejs_package_src}",
         package_name = "{package_name}",
         package_version = "{package_version}",
-        visibility = ["//visibility:public"],{maybe_indirect}{maybe_deps}
+        deps = {deps},
+        visibility = ["//visibility:public"],{maybe_indirect}
+    )
+"""
+
+_RUN_LIFECYCLE_HOOKS_TMPL = """run_binary(
```

Comment:
Nit: you can start the multiple line string with either a comment or a backslash so the nesting looks right